### ISSUE TITLE
MHdr header

### ISSUE THREAD

**Issue by aauffray** [scribbletune/scribbletune#87]
I'm trying to load MIDI files generated with Scribbletune into [@tonejs/midi](https://github.com/Tonejs/Midi). When I'm trying to do so, I'm always having this error from tone:
`Bad MIDI file.  Expected 'MHdr', got: ''`

After a little research, I found MHdr to be a header for MIDI files. I tried loading a scribbletune-generated MIDI file into FL Studio and immediately re-exporting it into a new MIDI file, then loading this new file into tonjs/midi. It worked.

My conclusion is Scribbletune doesn't add the MHdr header in its MIDI files. Is that it ? If so, is there a plan to add that into the module ?
Alternatively, since I'm trying to use tonejs/midi to edit BPM/tempo of my MIDI files, is there a plan to add BPM setting into Scribbletune in Node.js ?

I know I can switch to another MIDI generation module but I find Scribbletune excellent and I wish to keep using it

**Comment by walmik** [comment]
This is very interesting @aauffray, I believe Scribbletune does add the said header, in fact it uses jsmidgen which explicitly adds this. If that was not the case then FL studio itself wouldn’t recognize the midi generated by Scribbletune. Maybe something in tonejs midi needs additional stuff which Scribbletune doesn’t add (I ll look into this once I m back home - currently traveling)

I m curious to know how your usecase for tonejs midi. Maybe you re already aware but Scribbletune has tonejs as a dependency and allows you to [produce music](https://scribbletune.com/documentation/browser/browser-clip) or create [music applications like Johann](johann.scribbletune.com) in the browser.  [Here s one that lets you adjust the BPM](https://scribbletune.github.io/live/rpg8.html) with tonejs behind the scenes.

**Comment by aauffray** [comment]
I feel quite stupid.

After posting this, I tried displaying hexa for a scribbletune-generated MIDI and another MIDI generated via FL Studio (one that loads correctly into tonejs/midi)

Scribbletune:
`0000-0010:  4d 54 68 64-00 00 00 06-00 00 00 01-00 80 4d 54  MThd.... ......MT
0000-0020:  72 6b 00 00-01 24 00 90-50 7f 83 00-80 50 5a 00  rk...$.. P....PZ.
0000-0030:  90 52 7f 81-00 80 52 5a-00 90 4b 7f-81 00 80 4b  .R....RZ ..K....K`

FL Studio:
`0000-0010:  4d 54 68 64-00 00 00 06-00 01 00 02-00 60 4d 54  MThd.... .....`MT
0000-0020:  72 6b 00 00-00 13 00 ff-51 03 07 35-79 00 ff 58  rk...... Q..5y..X
0000-0030:  04 04 01 18-08 00 ff 2f-00 4d 54 72-6b 00 00 00  ......./ .MTrk...`

We can see MThd headers are displayed on both files. So I tried loading an older Scribbletune-generated file, just in case. Turns out I was trying to `fs.readFileSync` a file right after `await scribble.midi(clip, filePath)`, so the file existed but maybe wasn't fully dumped, explaining the header not found error. I thought awaiting scribble.midi made sure it was ok, but the function wasn't async.

The error was between chair and keyboard :) sorry for this one, great job on the module by the way !
